# Info
출처: https://nathanrosidi.medium.com/top-10-machine-learning-algorithms-for-beginner-data-scientists-aae78826712f

이미지


# 당뇨병 데이터세트 로드

In [1]:
from sklearn.datasets import load_diabetes
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# 데이터 세트 분할

In [2]:
# # 당뇨병 데이터세트를 로드합니다
diabetes = load_diabetes()
X, y = diabetes.data, diabetes.target

# 데이터세트를 학습 및 테스트 세트로 분할합니다
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 선형 회귀 모델 만들기 및 학습

In [3]:
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

# 예측 및 평가

In [4]:
# 테스트 세트 결과 예측
y_pred = model.predict(X_test)

# 평가지표 계산
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("Mean Squared Error:", mse)
print("R-squared:", r2)

Mean Squared Error: 2900.193628493482
R-squared: 0.4526027629719195


# 결과

이러한 결과는 선형 회귀 모델이 당뇨병 데이터 세트의 분산의 약 45%를 설명한다는 것을 나타냅니다. MSE는 평균적으로 예측이 실제 값에서 약 2900단위 떨어져 있음을 알려줍니다.